data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy

```diff
@@ -169,7 +169,7 @@ interface MyRepository {
         def encoded = encoder.buildQuery(AnnotationMetadata.EMPTY_METADATA, q)
 
         expect:
-        encoded.query.startsWith('SELECT restaurant_.`id`,restaurant_.`name`,restaurant_.`address_street`,restaurant_.`address_zip_code`,restaurant_.`hqaddress_street`,restaurant_.`hqaddress_zip_code` FROM')
```

Comment:
I think when there is 
```
@Relation(Relation.Kind.EMBEDDED)
private final Address address;
```
or 
```
@EmbeddedId
private PrimaryKey primaryKey;
```
then column should not be prefixed with `address_` or `primary_key_` and this was issue since while ago. This PR should fix it. Can cause some regression issue for existing apps, but the workaround would be adding `@MappedProperty` on embedded entity field.

Reply:
I don't think this is correct, and we can't cause regressions otherwise it has to go into next major version

Reply:
Yes, we can postpone it for 5.0